packages/serverpod/lib/src/server/session.dart

```diff
@@ -278,6 +278,52 @@ abstract class Session implements DatabaseAccessor {
       stackTrace: stackTrace,
     );
   }
+
+  /// Runs the closure [f] in isolation and undoing any changes in case it fails.
+  ///
+  /// This either reused the parent [transaction], or creates one itself.
```

Comment:
Split up the transaction / savepoint variants on this level, and then document the possible errors for each case.

Reply:
Update the comment accordingly.